electrum/submarine_swaps.py

```diff
@@ -541,3 +532,32 @@ def get_send_amount(self, recv_amount: Optional[int], *, is_reverse: bool) -> Op
         if is_reverse and send_amount is not None:
             send_amount += self.get_claim_fee()
         return send_amount
+
+    def get_swap_by_tx(self, tx):
+        txin = tx.inputs()[0]
+        # FIXME: not clear when TxInput has address
+        if hasattr(txin, 'address'):
```

Comment:
Pure `TxInput` never has `address`; only a `PartialTxInput` might. i.e. a "complete" tx will not have it.
In general, to know the address, we need some information about the output of the parent tx, which we often do not have.

Reply:
It might be interesting to look at `wallet.get_txin_address`; but it's not useful here.
https://github.com/spesmilo/electrum/blob/e00c998df346bd81a2de2720c3b01307d4a31eba/electrum/address_synchronizer.py#L139-L152

Reply:
this is not needed anymore, I found how to do it by comparing `funding_txid`